operators/src/processing/reprojection.rs

```diff
@@ -413,4 +667,229 @@ mod tests {
+        let qp = initialized_operator
+            .query_processor()
+            .unwrap()
+            .get_u8()
+            .unwrap();
+
+        let qs = qp
+            .raster_query(
+                QueryRectangle {
+                    bbox: output_bounds,
+                    time_interval,
+                    spatial_resolution,
+                },
+                &query_ctx,
+            )
+            .unwrap();
+
+        let _res = qs
+            .map(Result::unwrap)
+            .collect::<Vec<RasterTile2D<u8>>>()
+            .await;
+
+        // TODO: check against original data
+        //for t in res {
+        //    dbg!(&t.tile_information());
+        //}
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn raster_ndvi_3857() -> Result<()> {
+        let mut exe_ctx = MockExecutionContext::default();
+        let query_ctx = MockQueryContext::default();
+        let id = add_ndvi_data_set(&mut exe_ctx);
+
+        let output_shape: GridShape2D = [256, 256].into();
+        let output_bounds = BoundingBox2D::new_unchecked(
+            (-20_000_000., -20_000_000.).into(),
+            (20_000_000., 20_000_000.).into(),
+        );
+        let time_interval = TimeInterval::new_unchecked(1_388_534_400_000, 1_388_534_400_001);
+        // 2014-01-01
+
+        let gdal_op = GdalSource {
+            params: GdalSourceParameters {
+                data_set: id.clone(),
+            },
+        }
+        .boxed();
+
+        let projection = SpatialReference::new(
+            geoengine_datatypes::spatial_reference::SpatialReferenceAuthority::Epsg,
+            3857,
+        );
+
+        let initialized_operator = RasterOperator::boxed(Reprojection {
+            vector_sources: vec![],
+            raster_sources: vec![gdal_op],
+            params: ReprojectionParams {
+                target_spatial_reference: projection, // This test will do a identity reprojhection
+            },
+        })
+        .initialize(&exe_ctx)?;
+
+        let x_query_resolution = output_bounds.size_x() / output_shape.axis_size_x() as f64;
+        let y_query_resolution = output_bounds.size_y() / output_shape.axis_size_y() as f64;
+        let spatial_resolution =
+            SpatialResolution::new_unchecked(x_query_resolution, y_query_resolution);
+
+        let qp = initialized_operator
+            .query_processor()
+            .unwrap()
+            .get_u8()
+            .unwrap();
+
+        let qs = qp
+            .raster_query(
+                QueryRectangle {
+                    bbox: output_bounds,
+                    time_interval,
+                    spatial_resolution,
+                },
+                &query_ctx,
+            )
+            .unwrap();
+
+        let res = qs
+            .map(Result::unwrap)
+            .collect::<Vec<RasterTile2D<u8>>>()
+            .await;
+
+        // TODO: check against reference  data
+        for t in res {
+            dbg!(&t.tile_information());
+        }
```

Comment:
Same here.

Reply:
yes but i have to figure out how to generate the test data.

Reply:
e9b64a8